src/wayland/shell/xdg/mod.rs

```diff
@@ -240,12 +243,13 @@ macro_rules! xdg_role {
                     .unwrap_or(&self.current)
             }
 
-            /// Check if the state has pending changes.
+            /// Check if the state has pending changes that have
+            /// not been sent to the client.
             ///
-            /// This differs to just checking if the server pending
-            /// state is some in that it also check if a pending
-            /// state is different from the current server state.
-            fn has_pending_changes(&self) -> bool {
+            /// This differs to just checking if the [`server_pending`](#structfield.server_pending)
+            /// state is [`Some`] in that it also checks if a current pending
+            /// state is different from the [`current_server_state`](#method.current_server_state).
+            pub fn has_pending_changes(&self) -> bool {
```

Comment:
I'm realizing that this may be confusing to keep this name for the method, given it does not only checks the pending state. what about `has_unacked_changes` maybe ?

Reply:
There is a case where state can be accked but not yet committed, that would probably cause this to still return `true`.
I have to test out the newest version of this PR to make sure that it's still the case, tho.

Reply:
@PolyMeilex 

> There is a case where state can be accked but not yet committed, that would probably cause this to still return `true`. I have to test out the newest version of this PR to make sure that it's still the case, tho.

This should be covered [here](https://github.com/Smithay/smithay/pull/415/files#diff-e627a7774fb6d28266da99a8dd7f4b5d37af2e920cc7b84d94a169dff3ba7d94R242), or at least i thought that i covered that case by using the `last_acked` state if it is available in `current_server_state`. 

@vberger 

> I'm realizing that this may be confusing to keep this name for the method, given it does not only checks the pending state. what about `has_unacked_changes` maybe ?

I think `has_unacked_changes` could be missleading. The function checks if something has been configured using 'with_pending_state' on the server, but not sent to the client using 'send_configure'. So i think the nameing is appropriate, but on the other hand i am not the best at naming things XD

Reply:
> This should be covered here, or at least i thought that i covered that case by using the last_acked state if it is available in current_server_state.

Yeah, I know that, I was just mentioning that this should be taken into account as well when naming this method, but now that I look at the code it actually does not matter, as it only cares about ack.